### PR TITLE
Remove duplicate references from the referee survey export and cleanse data

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

When we duplicate an application form for apply again or at the end of the cycle, we also duplicate the references. This means that we have quite a few duplicate references in this export.

Additionally, we've currently got invalid data in our system as when I backfilled the `feedback_provided_at` as carried over applications did not have the required audits to populate the values.

## Changes proposed in this pull request

This PR does two things. 
- Marks carried over references as duplicates
- Populates missing feedback_provided_at values

## Guidance to review

Per commit is probably the easiest.

## Link to Trello card

https://trello.com/c/CL1ZO70r/3124-investigate-why-there-are-missing-dates-in-referee-survey-extract

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by stevehook — COMMENTED** [review]
Code makes sense but I've got a couple of questions about the approach...

I'm a bit worried about setting `feedback_provided_at` to `nil`. I can't see any code that obviously depends on this attribute being present so maybe it's fine. But does this mean there is data missing on apply again applications? (i.e. do providers/support/candidates need to see when the reference was provided?) Would it be possible to use some other attribute (or even a new dedicated attribute) to indicate that a reference was a duplicate and therefore not to be included in reporting?

**Comment by davidgisbey** [comment]
Hey Steve, the only place it's exposed at the moment is in the support UI. We don't actually show providers or candidates when it was provided. I did think about possibly adding a `duplicated` boolean to the form. I'll have a look into it today.

File: db/migrate/20210308172728_add_duplicate_boolean_to_references.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddDuplicateBooleanToReferences < ActiveRecord::Migration[6.0]
+  def change
+    add_column :references, :duplicate, :boolean, default: false
+  end
+end
```
Comment (stevehook) [inline db/migrate/20210308172728_add_duplicate_boolean_to_references.rb:5]: I'm guessing this will go away after rebasing as it's in a separate PR?

**Review by stevehook — APPROVED** [review]
This looks great. I've got a few minor suggestions inline.

**Pull request merged**